sky/clouds/cloud_registry.py

```diff
@@ -1,31 +1,65 @@
 """Clouds need to be registered in CLOUD_REGISTRY to be discovered"""
 
 import typing
-from typing import Optional, Type
+from typing import Callable, Dict, List, Optional, Type
 
 from sky.utils import ux_utils
 
 if typing.TYPE_CHECKING:
     from sky.clouds import cloud
 
 
-class _CloudRegistry(dict):
+class _CloudRegistry:
     """Registry of clouds."""
 
+    def __init__(self):
+        self.clouds: Dict[str, 'cloud.Cloud'] = {}
+        self.aliases: Dict[str, str] = {}
+
     def from_str(self, name: Optional[str]) -> Optional['cloud.Cloud']:
         if name is None:
             return None
-        if name.lower() not in self:
+        if name.lower() not in self.clouds:
             with ux_utils.print_exception_no_traceback():
                 raise ValueError(f'Cloud {name!r} is not a valid cloud among '
-                                 f'{list(self.keys())}')
-        return self.get(name.lower())
-
-    def register(self, cloud_cls: Type['cloud.Cloud']) -> Type['cloud.Cloud']:
-        name = cloud_cls.__name__.lower()
-        assert name not in self, f'{name} already registered'
-        self[name] = cloud_cls()
-        return cloud_cls
+                                 f'{list(self.clouds.keys())}')
+        return self.clouds[name.lower()]
+
+    def from_name_or_alias(self,
```

Comment:
Initially this is what I did, but there are cases where it's assumed that `from_str` will only work for the "canonical" name of the cloud.

For instance, if I hadn't made changes to `show_gpus`, updating `from_str` would cause `sky show-gpus --cloud k8s` to crash. Although this particular case is fixed by my changes, I'm worried there are other uses of from_str that rely on this behavior that I would be breaking.